native/c/zjb.cpp

```diff
@@ -296,12 +296,21 @@ int zjb_delete_by_jobid(ZJB *zjb, string jobid)
   return ZJBMPRG(zjb);
 }
 
-int zjb_submit(ZJB *zjb, string data_set, string &jobId)
+int zjb_submit(ZJB *zjb, string dsn_or_contents, string &jobId, bool is_contents)
```

Comment:
just a discussion point I'm curious about your opinion... we could also
1. pass in dsn or a pointer to contents on the `ZJB` struct
2. add another function that is `zjb_submit_dsn` which calls this one passing raw contents

what i'm thinking about is the `.cpp` `.hpp` layer being our programmatic APIs and what would be the easiest interface for consumers of the dll

Reply:
I 100% agree, I'll need to refactor the function a bit to have common code but it shouldn't be too bad. Will follow up on this comment once addressed 👍

Reply:
I'm going to go with option 2 because the first one would involve heap allocation for the data set contents (since the ZJB struct is in a C header)

Will resolve this soon ^^

Reply:
Addressed in 0ab0f31 👍

Reply:
cool stuff!